TESTING.md

```diff
@@ -0,0 +1,110 @@
+    # Call that should raise Error here:
+    should_not_be_a_list = ['Rebellious', 'List']
+```
+If the above function raised _SomeErrorType_, this test would pass.  Raising a different error or if nothing is raised, fails the test.  If you are not asserting an exception was raised, it should not be caught.
+
+
+* Use Parametrize to reduce test function count.  There is no need to copy/paste a test with a different set of input values.
+
+# Fixtures & Parametrization
+
+A fixture is best described as a modular resource that is set up, utilized, and then cleaned up automagically for the purposes of testing.
+
+As an example, we have two ``Rat`` fixtures used often:
+
+````python
+@pytest.fixture
+def rat_no_id_fx():
+    """
+    Returns: (Rescue): Rescue test fixture without an api ID
+
+    """
+    return Rats(None, "noIdRat")
+````
+
+````python
+@pytest.fixture(params=[("myPcRat", Platforms.PC, UUID("dead4ac0-0000-0000-0000-00000000beef")),
+                        ("someXrat", Platforms.XB, UUID("FEED000-FAC1-0000-0000900000D15EA5E")),
+                        ("psRatToTheRescue", Platforms.PS,
+                         UUID("FEE1DEA-DFAC-0000-000001BADB001FEED"))],
+                )
+def rat_good_fx(request) -> Rats:
+    """
+    Testing fixture containing good and registered rats
+    """
+    params = request.param
+    myRat = Rats(params[2], name=params[0], platform=params[1])
+    return myRat
+````
+
+The first returns a simple, uninitialized rat object, while the second one is parametrized to run any test its used in three times, initialized three times; once for each platform.
+
+Most importantly, these fixtures make module calls to their appropriate sub objects, and can be used without breaking scope in other unit tests, and are disposed so that each iteration of parameters uses a 'clean' object.
+
+Here's an example of a valid and complete test:
+```python
+@pytest.mark.asyncio
+    @pytest.mark.parametrize("garbage", [22.1, -42, 42, 0, False, True])
+    async def test_find_rat_by_name_bad_type(self, garbage):
+        """
+        Verifies that attempting to throw garbage at Rats.search() raises the proper exception
+        """
+        with pytest.raises(TypeError):
+            await Rats.get_rat_by_name(name=garbage)
+
+        with pytest.raises(TypeError):
+            await Rats.get_rat_by_name(name="foo", platform=garbage)
+```
+
+# Marking Tests
+
+Tests are categorized by module using ``pytest.mark`` so that specific tests may be excluded.  Each module's decorator is set using the ``pytestmark`` global:
+
+```python
+# Replace 'category' with an unused one for your module, after it has been added to pytest.ini
+pytestmark = pytest.mark.category
+```
+
+Test categories are listed in [pytest.ini](./pytest.ini).  It may be necessary to run tests locally without using the ``database`` or ``api`` markers if you do not have these available.  This can be done by running py.test in the following manner:
```

Comment:
```md
...markers if you do not have these available...
```
perhaps replace that with "if those test requisites are not available in your dev environment"?